app/javascript/controllers/multiple_select_controller.js

```diff
@@ -3,8 +3,61 @@ import TomSelect from 'tom-select'
 
 export default class extends Controller {
   static targets = ['select']
+  static values = {
+    options: Array,
+    optionGroups: Array,
+    optgroupField: String,
+    labelField: String,
+    defaultSelected: Array
+   }
 
   connect () {
+    const isGrouped = this.hasOptionsValue &&
+                        this.hasOptionGroupsValue &&
+                        this.hasOptgroupFieldValue &&
+                        this.hasLabelFieldValue
+    if (isGrouped) {
+      /* eslint-disable no-new */
+      return new TomSelect(this.selectTarget, {
+        plugins: {
+          remove_button: {
+            title: 'Remove this item',
+            className: 'text-primary p-1 mx-2 deactive-bg rounded-circle',
+            label: `<i class="lni lni-close p-1"></i>`
+          }
+        },
+        options: this.optionsValue,
+        optgroups: this.optionGroupsValue,
+        optgroupField: this.optgroupFieldValue,
+        labelField: this.labelFieldValue,
+        items: this.defaultSelectedValue,
+        searchField: [this.labelFieldValue, this.optgroupFieldValue],
+        render: {
+          option: function(data, escape) {
+            return `
+                <div class='d-flex gap-1'>
+                  <span class='d-flex gap-1'>${escape(data.label)}</span>
```

Comment:
b/c of a quirk w/ the highlighting implementation in tomselect, using `gap-1` and `d-flex` to keep everything on the same line. My preference exists for maintaining spaces and adding one extra space to the right than removing spaces altogether. 

I think this implementation is good enough, though if we choose to remove highlighting altogether, we would likely avoid this being a problem.